scripts/api/data/metadatablocks/archival.tsv

```diff
@@ -0,0 +1,19 @@
+#metadataBlock	name	dataverseAlias	displayName	blockURI												
+	archive		Archival Metadata													
+#datasetField	name	title	description	watermark	fieldType	displayOrder	displayFormat	advancedSearchField	allowControlledVocabulary	allowmultiples	facetable	displayoncreate	required	parent	metadatablock_id	termURI
+	archiveSubmitToArchivalAppraisal	Archival Appraisal	An assessment whether the dataset should be submitted for archival appraisal		text	0	#VALUE	TRUE	FALSE	FALSE	FALSE	TRUE	FALSE		archive	
+	archiveArchivedFrom	Archived from	A date (YYYY-MM-DD) from which the dataset is archived	YYYY-MM-DD	date	2	#VALUE	TRUE	FALSE	FALSE	FALSE	FALSE	FALSE		archive	
+	archiveArchivedFor	Retention period	The retention period for which the archived dataset is to be kept by the holding archive		text	3	#VALUE	TRUE	TRUE	FALSE	FALSE	FALSE	FALSE		archive	
+	archiveHoldingArchive	Archived at Holding Archive	The holding archive where the dataset is archived		text	4	#VALUE	FALSE	FALSE	TRUE	FALSE	FALSE	FALSE		archive	https://schema.org/holdingArchive
+	archiveArchivedAt	Archived at URL	URL to holding archive where the dataset is archived	URL	url	5	<a href='#VALUE'>#VALUE</a>	FALSE	FALSE	TRUE	FALSE	FALSE	FALSE		archive	https://schema.org/archivedAt
+#controlledVocabulary	DatasetField	Value	identifier	displayOrder												
+	archiveArchivedFor	1 year	1_year	0												
+	archiveArchivedFor	3 years	3_years	1												
+	archiveArchivedFor	5 years	5_years	2												
+	archiveArchivedFor	10 years	10_years	3												
+	archiveArchivedFor	20 years	20_years	4												
+	archiveArchivedFor	Indefinitely	indefinitely	5												
+	archiveArchivedFor	Unknown	unknown	6												
+	archiveSubmitToArchivalAppraisal	Yes	yes	0												
+	archiveSubmitToArchivalAppraisal	No	no	1												
+	archiveSubmitToArchivalAppraisal	Unknown	unknown	2												
```

Comment:
Is this the boolean? Yes, no, or unknown? As discussed here (and in person in Mexico):

- #7961

Reply:
This PR only implements a controlled vocabulary list, because the boolean is not yet in the source code. This PR, #10480 , has grown stale, I think, but if possible when 10480 is further refined and merged, then the PR for this archival.tsv metadata block should adopt the boolean for this field specifically, because yes, as we discussed in Mexico, this is exactly the use case why we need the boolean to properly handle the data type in Dataverse. My thought at the moment is to merge the archival.tsv and then when 10480 is merged send in a new PR to adopt it for archival.tsv.

what do you think?

Reply:
Yes, what you've done is similar (I think) to other quasi booleans we've defined in other metadata blocks, given the tools we have.

As of that boolean PR, I did at least leave a comment just now: https://github.com/IQSS/dataverse/pull/10480#discussion_r1636952509

And yeah, especially if archival.tsv is marked as experimental in the guides, I imagine we could update it to take advantage of new functionality (like booleans) as they come along.

Reply:
Cool! Thanks for your comments.

The mapping of the boolean from frontend to backend could work like this:

Yes = true
No = false
Unknown = null